### ISSUE TITLE
pnpm install not expending modules as npm3

### ISSUE THREAD

**Issue by idanh** [rstacruz/pnpm#346]
### pnpm version:
0.29.1

### Code to reproduce the issue:
npm install
ls ./node_modules

pnpm install
ls ./node_modules

You can see that the difference in expansion of the modules.

### Expected behavior:
Should be exactly the same

### Actual behavior:
The transitive behavior of pnpm which mirrors an older npm version.

### Additional information:
The issue that lead me to check that phenomena is that gulp, used in a specific way (requiring from a module) will not work with an older version (< 3) of npm. So for now, I can either use pnpm or gulp

**Comment by zkochan** [comment]
It seems to me like a misuse of an npm3 feature not intended to be used. If a package is required, why not include it in the package's dependencies? In that case it would be in the root of `node_modules`.

Can you provide a code example? What do you mean by requiring gulp from a module?

**Comment by idanh** [comment]
@zkochan, it may as well be a misuse but it is not mine. I cannot supply a sample code right now, but the scenario is as follows:

1. install gulp-cli globally (npm i gulp-cli -g)
2. create a module my-gulp-module with package.json as following: "gulp": "^3.9.1"
3. create another module my-app with package.json as following: "my-gulp-module": "../my-gulp-module"
4. in the second module (my-app) add gulpfile.js with a simple task that echos something.

Do not add gulp dependency to my-app, it should be implicitly derived from my-gulp-module (as it's explicitly specified in it's package.json)

If you're executing 'npm install' in my-app, everything will work.
If you're using 'pnpm install', the error you are probably going to get is that "gulp is not found"

Now, I'm not saying that it's *your* bug, I'm only giving a concrete example of a misbehaviour from 2 different setups. 

I also personally think that pnpm should mirror npm in the way it expends the packages so that it can be a drop-in replacement for npm

**Comment by zkochan** [comment]
Lets see what the other maintainers will say. `my-app` clearly has to depend on `gulp` if it uses it in code... and seems like you agree on that. It just happens to be that it works on npm3.

From the other side, this shouldn't be hard to "fix" on the pnpm side...

@rstacruz, @iamstarkov, @andreypopp what do you think?

**Comment by andreypopp** [comment]
I think what npm v3 does is wrong: it makes it safer to rely on
dependencies of dependencies which is never a good thing (consider the case
when such dependency changes — because you don't specify the exact version
it might break your app). pnpm does the right thing, I believe.

**Comment by iamstarkov** [comment]
I don't think this issue need to be fixed on pnpm side. if you need to reuse local packages, there is `p/npm link` for that

**Comment by idanh** [comment]
@andreypopp , I've verified that npm3 handles conflicts well (it preserve the transitive relation where there's a conflict).
So, in all cases, npm3 behaves well. It won't break the app and it's consistent with earlier implementations.

@iamstarkov, Reuse of local packages is not the issue, the expansion of modules is the side-effect of compiling my concrete example with pnpm.

I'll close this issue. It seems there's a consensus with not aligning with npm3. I suggest that you check if there's a bigger effect to this than what I've demonstrated here. As for my problem, I'll create a post-script to flatten those dependencies so I could continue using pnpm.

Thanks!
